### DISCUSSION TITLE
datamodule for seasonet dataset

### DISCUSSION THREAD

**Discussion by juanmanuelrq** [torchgeo/torchgeo#2076] [Ideas]
How can I create a 

LightningDataModule https://torchgeo.readthedocs.io/en/latest/api/datamodules.html#geospatial-datamodules 

of the seasonet dataset https://torchgeo.readthedocs.io/en/latest/api/datasets.html#seasonet

best regars

**Comment by adamjstewart** [comment]
Hi @juanmanuelrq! We don't yet have great documentation on how to create your own data module, but it's actually one of the easiest things to contribute if you want to open a PR! 

SeasoNet is a pretty standard dataset with standard train/val/test splits. The simplest working data module might be as simple as:
```python
from torchgeo.datamodules import NonGeoDataModule
from torchgeo.datasets import SeasoNet

class SeasoNetDataModule(NonGeoDataModule):
    def __init__(self, batch_size: int = 64, num_workers: int = 0, **kwargs: Any) -> None
        super().__init__(SeasoNet, batch_size, num_workers, **kwargs)
```
You can also add custom mean/std dev for normalization, custom data augmentation, custom args, etc. See `torchgeo/datamodules` for dozens of examples. 

The `NonGeoDataModule` base class is defined in `torchgeo/datamodules/geo.py` and may give you an idea of what features are available. An important thing to explain: you can use `train_aug`, `val_aug`, `test_aug`, etc. to define data augmentation specific to each step, or `aug` to define data augmentation for all steps.

Let me know if you make any progress on this. We would love to include this in the 0.6.0 release if you want to open a PR. @dkosm @briktor contributed the original dataset and may also be able to help, although I always prioritize new contributors 😄
